### ISSUE TITLE
False positive of DMI_RANDOM_USED_ONLY_ONCE for random used at least twice

### ISSUE THREAD

**Issue by DanySK** [spotbugs/spotbugs#3830]
**Describe the bug**
The detector `DMI_RANDOM_USED_ONLY_ONCE` flags code as:

> `Random object created and used only once`

even when the `java.util.Random` instance is used multiple times before being discarded.

The documentation of the rule explicitly says:

> This code creates a `java.util.Random` object, **uses it to generate one random number**, and then discards the `Random` object.

(emphasis mine)

In the attached example, the `Random` instance is used several times in a loop, yet the warning is still reported. This is either:

* a false positive according to the documented condition (“one random number”), or
* a documentation bug: the description does not match what the detector actually checks (short-lived local `Random`).

Right now, the message and behaviour are inconsistent.

---

**To Reproduce**

1. Create a project with the following class.
2. Run SpotBugs with rule `DMI_RANDOM_USED_ONLY_ONCE` enabled.
3. Observe that `DMI_RANDOM_USED_ONLY_ONCE` is reported on the constructor.

Please provide a code example, if applicable. Preferably a minimal, verifiable code example.

```java
import java.util.HashSet;
import java.util.Random;
import java.util.Set;

public class LogicsImpl {

    private final int gridSize;
    private final Set<Pair<Integer, Integer>> set = new HashSet<>();

    public LogicsImpl(final int gridSize) {
        this.gridSize = gridSize;
        final Random random = new Random();
        while (set.size() < 3) {
            // Random is used multiple times here
            this.set.add(new Pair<>(random.nextInt(gridSize), random.nextInt(gridSize)));
        }
    }

    private record Pair<A, B>(A first, B second) {}
}
```

SpotBugs reports something like:

> DMI: Random object created and used only once in new LogicsImpl(int)

even though `random.nextInt(...)` is called multiple times.

---

**Expected behavior**
One of the following:

1. **If the rule is meant to be literal (“one random number”)**
   Then the detector should only fire when there is exactly one call to a `Random` method producing a random value for that instance (e.g. `new Random().nextInt(...)`).
   In that case, the example above should **not** be flagged.

2. **If the rule is meant to be about lifetime/scope (“short-lived local Random”)**
   Then the documentation and message should be updated to reflect that. For example, something along the lines of:

   * “This code creates a short-lived `java.util.Random` object that is used only within this method/constructor and then discarded…”
     and remove or rephrase “uses it to generate one random number”.

In either case, the current wording (“uses it to generate one random number, and then discards the Random object”) does not match the behaviour seen in the example.

---

**Environment**
[FILL IN: replace the placeholders below with your actual environment]

* Java Compiler:

`javac` from the Java distribution listed below. `javac 21.0.8`

* Java version:
```
openjdk version "21.0.8" 2025-07-15
OpenJDK Runtime Environment (build 21.0.8+9)
OpenJDK 64-Bit Server VM (build 21.0.8+9, mixed mode, sharing
```

* SpotBugs Version: 4.9.8
* How do you use SpotBugs: Gradle Plugin
* Plugin version: SpotBugs Gradle Plugin 6.4.6
* Environment version for Plugin: Gradle 9.2.1

---

**Additional context**
From a user’s perspective, the current behavior looks like a false positive because the documented condition (“uses it to generate one random number”) is not met.
